src/frontend/components/StudentLiveControlBar/index.spec.tsx

```diff
@@ -0,0 +1,251 @@
+
+import {
+  LivePanelItem,
+  useLivePanelState,
+} from 'data/stores/useLivePanelState';
+import { LiveModeType, liveState } from 'types/tracks';
+import { videoMockFactory } from 'utils/tests/factories';
+import { wrapInIntlProvider } from 'utils/tests/intl';
+
+import { StudentLiveControlBar } from '.';
+import { wrapInRouter } from 'utils/tests/router';
+
+const mockEmptyVideo = videoMockFactory();
+const mockRunningJitsiWithXMPP = videoMockFactory({
+  id: '5cffe85a-1829-4000-a6ca-a45d4647dc0d',
+  live_type: LiveModeType.JITSI,
+  live_state: liveState.RUNNING,
+  xmpp: {
+    bosh_url: 'https://xmpp-server.com/http-bind',
+    websocket_url: null,
+    conference_url:
+      '870c467b-d66e-4949-8ee5-fcf460c72e88@conference.xmpp-server.com',
+    prebind_url: 'https://xmpp-server.com/http-pre-bind',
+    jid: 'xmpp-server.com',
+  },
+});
+
+describe('<StudentLiveControlBar />', () => {
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('renders no button', () => {
+    useLivePanelState.setState({});
+
+    render(
+      wrapInRouter(
+        wrapInIntlProvider(<StudentLiveControlBar video={mockEmptyVideo} />),
+      ),
+    );
+
+    expect(
+      screen.queryByRole('button', { name: 'Show apps' }),
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('buttont', { name: 'Show chat' }),
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('button', {
+        name: 'Send request to join the discussion',
+      }),
+    ).not.toBeInTheDocument();
+  });
+
+  it('renders apps wrapper button', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.APPLICATION],
+    });
+
+    render(
+      wrapInRouter(
+        wrapInIntlProvider(<StudentLiveControlBar video={mockEmptyVideo} />),
+      ),
+    );
+
+    screen.getByRole('button', { name: 'Show apps' });
+    expect(
+      screen.queryByRole('buttont', { name: 'Show chat' }),
+    ).not.toBeInTheDocument();
+    expect(
+      screen.queryByRole('button', {
+        name: 'Send request to join the discussion',
+      }),
+    ).not.toBeInTheDocument();
+  });
+
+  it('renders chat wrapper button', () => {
+    useLivePanelState.setState({
+      availableItems: [LivePanelItem.CHAT],
+    });
+
+    render(
+      wrapInRouter(
+        wrapInIntlProvider(<StudentLiveControlBar video={mockEmptyVideo} />),
+      ),
+    );
+
+    expect(
+      screen.queryByRole('button', { name: 'Show apps' }),
+    ).not.toBeInTheDocument();
+    screen.getByRole('button', { name: 'Show chat' });
+    expect(
+      screen.queryByRole('button', {
+        name: 'Send request to join the discussion',
+      }),
+    ).not.toBeInTheDocument();
+  });
+});
+
+describe('<StudentLiveControlBar /> leave/join discussion wrapper', () => {
+  it('does not render wrapper without xmpp available in video', () => {
+    useLivePanelState.setState({});
+
+    const mockVideo = { ...mockRunningJitsiWithXMPP, xmpp: null };
```

Comment:
In this describe block why not creating a video for each `it` ? They all have a special case, you can configure a dedicated video for each of them.